corehq/apps/data_cleaning/tests/mixins.py

```diff
@@ -0,0 +1,35 @@
+import uuid
+from abc import ABC, abstractmethod
+
+from corehq.apps.es.case_search import case_search_adapter
+from corehq.apps.es.client import manager
+from corehq.pillows.case_search import CaseSearchReindexerFactory
+from corehq.util.test_utils import create_and_save_a_case
+
+
+class CaseDataTestMixin(ABC):
+
+    @property
+    @abstractmethod
+    def case_type(self):
+        raise NotImplementedError(
+            "Subclasses must define a case_type property."
+        )
+
+    def make_case(self, domain, case_properties, index=None, case_type=None):
+        case_properties = case_properties or {}
+        case_id = case_properties.pop('_id')
+        case_type = case_properties.pop('case_type', case_type or self.case_type)
```

Comment:
Feel free to ignore this suggestion because it is just personal preference.

On one hand I like how flexible this is and how `NotImplementedError` standardizes the way that the `case_type` property has been set on subclasses. On the other hand, in all the usages `case_type` is passed in `case_properties`, which seems intuitive to me, but unless there are already usages that pass `case_type` as a parameter, or require falling back to the `case_type` class property, then I would lean towards simplifying this by removing the additional options. i.e.

```python
        case_type = case_properties.pop('case_type')
```
and dropping the `case_type` class property.

Reply:
yep that makes sense. updated!

Reply:
i originally took this code from some other ES case data tests, which had that old fallback